aws/resource_aws_api_gateway_integration.go

```diff
@@ -160,8 +172,8 @@ func resourceAwsApiGatewayIntegrationCreate(d *schema.ResourceData, meta interfa
 		RequestParameters:   aws.StringMap(parameters),
 		RequestTemplates:    aws.StringMap(templates),
 		Credentials:         credentials,
-		CacheNamespace:      nil,
-		CacheKeyParameters:  nil,
+		CacheNamespace:      aws.String(d.Get("resource_id").(string)),
```

Comment:
I just updated the pull request with your feedback. I did some testing and using a user-defined cache namespace works too. So, I added a field named `cache_namespace`.